src/query/interpret/eval.hpp

```diff
@@ -687,7 +715,12 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
 
   TypedValue Visit(MapLiteral &literal) override {
     TypedValue::TMap result(ctx_->memory);
-    for (const auto &pair : literal.elements_) result.emplace(pair.first.name, pair.second->Accept(*this));
+    for (const auto &pair : literal.elements_) {
+      result.emplace(pair.first.name, pair.second->Accept(*this));
+    }
+
+    ctx_->property_lookups_cache.clear();
```

Comment:
Hmmm, that is correct, I didn't think of that use case but only optimizing since we did fetch property by property. Seems like it could furthermore speed things up

Reply:
This is a really interesting idea, but right now it goes beyond the scope of this feature (based on client requests).

Reply:
Can you add a comment here for some future improvement?

Reply:
Done